Makefile

```diff
@@ -222,38 +222,9 @@ check: check-bench
 	# as default, and test a minimal set of features with only one backend
 	# at a time.
 	cargo check --manifest-path lib/runtime/Cargo.toml
-	# Check some of the cases where deterministic execution could matter
-	cargo check --manifest-path lib/runtime/Cargo.toml --features "deterministic-execution"
-	cargo check --manifest-path lib/runtime/Cargo.toml --no-default-features \
-		--features=default-backend-singlepass,deterministic-execution
-	cargo check --manifest-path lib/runtime/Cargo.toml --no-default-features \
-		--features=default-backend-llvm,deterministic-execution
-	cargo check --release --manifest-path lib/runtime/Cargo.toml
```

Comment:
Note: After a deep analysis I disabled all the unuseful checks, since only the default backends are mutually exclusive. For the rest we can check all the same features at once.

Reply:
It's still useful to check things like `debug`  in both states with everything else enabled because that has a very high chance of breaking due to our lints, which may be an argument for disabling the lints.

The other features can interact, too, it's just less likely.  My opinion is that the cost of a breakage in the build is very expensive in terms of developer time and resyncing on master and the cost to check it in CI is very relatively cheap so I don't like this trade off.  My opinion is that we should test all valid feature sets in CI.  What `default-backend-singlepass,singlepass,cranelift,llvm,cache,debug,deterministic-execution` says to me is that using any default backend but singlepass with other compilers enabled isn't supported, you must use the debug flag and deterministic execution, and caching because we don't test any other uses, therefore we won't even know if they don't work